src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -31,7 +30,7 @@ internal abstract partial class AbstractProject : IVisualStudioHostProject
 
         private readonly ProjectId _id;
         private readonly string _language;
-        private readonly IVsHierarchy _hierarchy;
+        private readonly IVsHierarchy _hierarchyDoNotAccessDirectly;
```

Comment:
Why not, since it's readonly and therefore not mutating or anything? Should this just become an autoprop like you did for some others?